app/src/main/java/io/capibaras/abcall/ui/viewmodels/LoginViewModel.kt

```
@@ -66,40 +62,35 @@ class LoginViewModel(
                     } else {
                         val errorBody = response.errorBody()!!.string()
                         val jsonObject = JSONObject(errorBody)
                         val message = jsonObject.getString("message")
-                        ErrorUIState.Error(message = message)
+                        stateMediator.setErrorState(ErrorUIState.Error(message = message))
                     }
                 }
             } catch (e: IOException) {
-                errorUIState = ErrorUIState.Error(R.string.error_network)
+                stateMediator.setErrorState(ErrorUIState.Error(R.string.error_network))
             } catch (e: Exception) {
```

detekt

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)